Extension/src/common.ts

```diff
@@ -292,6 +292,17 @@ export function isOptionalArrayOfString(input: any): input is string[]|undefined
     return input === undefined || isArrayOfString(input);
 }
 
+export function resolveCachePath(input: string, additionalEnvironment: {[key: string]: string | string[]}): string {
+    let resolvedPath: string = (process.platform === 'win32') ? process.env.USERPROFILE : process.env.HOME;
+    if (!input || input === "") {
```

Comment:
"" (Empty string) should already be a [falsy](https://basarat.gitbooks.io/typescript/docs/javascript/truthy.html). Where are you doing the check to see if specified path is invalid?

E.g. just
`if (!input)`

Reply:
The actual verification if any path exists will continue to remain in the C++ code.